ingestion-beam/src/main/java/com/mozilla/telemetry/util/DynamicPathTemplate.java

```diff
@@ -142,7 +142,7 @@ public String replaceDynamicPart(List<String> values) {
       // This regex matches either of these patterns,
       // capturing the name of the placeholder as group 1
       // and the default value (or null if no default is provided) as group 2.
-      "\\$\\{(\\p{Alnum}+)(?::-(\\p{Alnum}+))?}"
+      "\\$\\{([A-Za-z0-9-_]+)(?::-([A-Za-z0-9-_]+))?}"
```

Comment:
Attribute keys and values can be any string, so I think we should make this a lot broader, like: `"\\$\\{([^${:}]+)(?::-([^${:}]*))?}"`. This allows anything but special characters, and lets the default be empty.

I think we also need to expand this to throw an exception if there are any `${` that don't map to a placeholder, but that can be filed as an issue for later.

Reply:
> I think we also need to expand this to throw an exception if there are any ${ that don't map to a placeholder, but that can be filed as an issue for later.

I'm not sure I fully understand the nuance you're getting at here. Would you be willing to file an issue with the specifics?

Reply:
yeah, i'll file an issue